bytestring_splitter/__init__.py

```diff
@@ -379,6 +378,182 @@ def _parse_message_meta(message_item):
         return BytestringSplitter.Message(message_name, message_class, message_length, kwargs)
 
 
+class HeaderMetaDataMixinBase:
+    """
+    A baseclass for mixins that work by serializing metadata about a bytestring by adding bytes to the
+    start of said bytestring and deserialize that data at other times by removing those same bytes.
+    """
+
+    def __call__(self, splittable, *args, **kwargs):
+        setattr(self, f'input_{self.METADATA_TAG}', kwargs.pop(self.METADATA_TAG, None))
+        splittable = self.strip_metadata(splittable)
+        splitter = super().__call__(splittable, *args, **kwargs)
+        return splitter
+
+    @classmethod
+    def _get_ordered_mixin_chain(cls, reversed=False):
+        """
+        returns mixins inheriting from HeaderMetaDataMixinBase in MRO order
+        for the purpose of removing or adding bytes in the correct order
+        """
+
+        mixins = [
+            kls for kls in cls.__mro__ if
+            issubclass(kls, HeaderMetaDataMixinBase)
+            and not issubclass(kls, BytestringSplitter) and kls
+            is not HeaderMetaDataMixinBase and kls
+            is not cls
+            ]
+
+        # if an implementer creates a splitter by directly subclassing this baseclass and BytestringSplitter
+        if not mixins and (issubclass(cls, HeaderMetaDataMixinBase) and cls not in mixins):
```

Comment:
Wait a second. `not mixins` is True is mixins are empty, and they are empty - `cls` will always be not in mixins anyway!

Reply:
Was it meant `if mixins and ...`?

Reply:
Yes you’re right.